Include/object.h

```diff
@@ -124,12 +124,45 @@ typedef struct {
 #define Py_TYPE(ob)             (_PyObject_CAST(ob)->ob_type)
 #define Py_SIZE(ob)             (_PyVarObject_CAST(ob)->ob_size)
 
+
+/* [RFC] Should we enable Immortal Instances by Default? */
+#define Py_IMMORTAL_INSTANCES
+
+/* Immortalizing causes the instance to not participate in reference counting.
+ * Thus, an immortal object will be kept alive until the runtime finalization.
+ * This avoids an unnecessary copy-on-write for applications that share
+ * a common python heap across many processes. */
+#ifdef Py_IMMORTAL_INSTANCES
+
+/* The GC bit-shifts refcounts left by two, and after that shift we still
+ * need this to be >> 0, so leave three high zero bits (the sign bit and
+ * room for a shift of two.) */
+static const Py_ssize_t kImmortalBit = 1L << (8 * sizeof(Py_ssize_t) - 4);
+
+static inline int _Py_IsImmortal(PyObject *op)
+{
+    return (op->ob_refcnt & kImmortalBit) != 0;
```

Comment:
Could you move these functions to the private API as well (`Include/internal/pycore_gc.h`)?

Reply:
Can do!